src/ip_data.py

```diff
@@ -78,10 +80,115 @@ def interleave(iters: list[Iterator[T]]) -> Iterator[T]:
 
 
 def repeat(genrec: Callable[[], T], times: int) -> Iterator[T]:
-    for i in range(times):
+    for _ in range(times):
         yield genrec()
 
 
+class FileWriter(ABC):
+    def __init__(self, base_filename: str):
+        self.base_filename = base_filename
+
+    @abstractmethod
+    def write_access(self, generator: Iterator[List[Any]]) -> None:
+        pass
+
+    @abstractmethod
+    def write_people(self, generator: Iterator[List[Any]]) -> None:
+        pass
+
+    @abstractmethod
+    def write_subnet(self, generator: Iterator[List[Any]]) -> None:
+        pass
+
+    @abstractmethod
+    def init_tables(self) -> None:
+        pass
+
+    @abstractmethod
+    def __enter__(self):
+        return self
+
+    @abstractmethod
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
```

Comment:
why is this an abstractmethod? Not required, contextmanagers don't work like that, please follow what I said before

Reply:
removed